lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -204,15 +206,17 @@ def cluster_resource_discoverer
       )
     end
 
-    def ejson_provisioner
-      @ejson_provisioner ||= EjsonSecretProvisioner.new(
-        namespace: @namespace,
-        context: @context,
-        template_dir: @template_dir,
-        logger: @logger,
-        statsd_tags: @namespace_tags,
-        selector: @selector,
-      )
+    def ejson_provisioners
+      @ejson_provisioners ||= @template_dirs.map do |template_dir|
+        EjsonSecretProvisioner.new(
```

Comment:
It seemed more straightforward to leave the provisioner itself alone and simply create as many instances as there are template directories (which will conceivably be a small number). We can then aggregate all the resources together on [line 274](https://github.com/Shopify/kubernetes-deploy/pull/514/files#diff-4c33bbb177fdf882c41d892f3bde9570R274) and proceed as normal.

An alternative would be to separate certain common methods out into the `EjsonSecretProvisioner` metaclass to avoid having to make excessive calls to e.g. `ejson-keys`. In practice, I don't think this is worth it, but I can be swayed easily if others feel strongly.